Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -239,9 +239,24 @@ auto compilationMessageData(convertMessages(
         m_library.label = label;
 }
 
-id<MTLFunction> ShaderModule::getNamedFunction(const String& name) const
+id<MTLFunction> ShaderModule::getNamedFunction(const String& name, const HashMap<String, float>& keyValueReplacements) const
 {
-    return [m_library newFunctionWithName:name];
+    if (!keyValueReplacements.size())
+        return [m_library newFunctionWithName:name];
+
+    MTLFunctionConstantValues *constantValues = [MTLFunctionConstantValues new];
+    for (auto& kvp : keyValueReplacements) {
+        // FIXME: specify actual types from the compiler information
+        [constantValues setConstantValue:&kvp.value type:MTLDataTypeFloat withName:kvp.key];
+    }
+
+    NSError *error;
+    id<MTLFunction> result = [m_library newFunctionWithName:name constantValues:constantValues error:&error];
+
+    if (error)
+        WTFLogAlways("MSL compilation error: %@", error);
```

Comment:
I think we should do this properly (or add a FIXME and a bug link). This is supposed to push an uncategorized error to the current error scope.

Reply:
I'll add a bug link for proper logging throughout WebGPU.framework